src/main/java/uk/gov/homeoffice/digital/sas/timecard/model/TimeEntry.java

```diff
@@ -0,0 +1,70 @@
+package uk.gov.homeoffice.digital.sas.timecard.model;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import org.hibernate.annotations.Type;
+import uk.gov.homeoffice.digital.sas.jparest.annotation.Resource;
+import uk.gov.homeoffice.digital.sas.jparest.models.BaseEntity;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.validation.constraints.NotNull;
+import java.util.Date;
+import java.util.UUID;
+
+@Resource(path = "time-entry")
+@Entity(name = "time_entry")
+@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
+@NoArgsConstructor
+@Getter
+@Setter
+public class TimeEntry extends BaseEntity {
+
+    private int version;
+
+    @NotNull
+    @Column(name = "owner_id")
+    private int ownerId;
+
+    @Type(type="uuid-char")
+    @Column(name = "time_period_type_id")
+    private UUID timePeriodTypeId;
+
+    @ManyToOne
+    @JoinColumn(name="time_period_type_id", referencedColumnName = "id", unique = true, nullable = false, insertable = false, updatable = false)
+    @JsonIgnore
+    private TimePeriodType timePeriodType;
+
+    @Column(name = "shift_type")
+    private String shiftType;
+
+    @NotNull
+    @Column(name = "actual_start_time")
+    private Date actualStartTime;
```

Comment:
Maybe we should be using one of java.time classes (LocalDateTime?) instead of java.util.Date

Reply:
Yeah I would also prefer LocalDateTime, easier to work with imo.

Reply:
I'm going to add Temporal type that does not support LocalDateTime and more over these fields should be updated automatically and there is no way to convert into LocalDateTime. Please do some research on @CreationTimestamp with @TemporalType. Hope that helps you

Reply:
Good point, i think when we start implementing other logic around this we can just simply convert to `LocalDateTime` type within the actual methods to do any date/time related operations. Model wise I think Date should be fine with your temporal type.

Reply:
@sureshmanga1 if the entities are being generated from database schema using a tool, maybe it can be configured to use java.time types instead of util.Date and Temporal?

Since JPA 2.2 the mapping of java.time types is supported natively. I believe we using Hibernate 5.6.* which is compatible with JPA 2.2

Further details and code samples can be found here:
https://www.baeldung.com/jpa-java-time#after-java-8

Reply:
Wow Thank you @taomoh, I will implement it